ts-jest not created when it does not exist and then coverage crash

- Issue

test run & pass but i've got an error (probably when coverage is ran)
"No such file or directory" when running jest



- Expected behavior

ts-jest should create a folder `ts-jest` in `cacheDirectory` if it does not exists


i think i'm expriencing the same issue
```
node v7.7.3
npm  v4.1.2

script:{
    "test": "jest --config jestconfig.json --runInBand",
    "test:watch": "npm test -- --watch",
    "test:cover": "npm test -- --coverage --no-cache"
}
```

```
>npm test

> react-typescript2-jest@0.0.1 test C:\Workspace\GitHub\TeBeCo\webpack-seeds\react-typescript2-jest
> jest --config jestconfig.json --runInBand

 PASS  src\somePath\xx.spec.ts
  given xxxxx
    √ should be true (16ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        0.766s, estimated 1s
Ran all test suites.
Error: ENOENT: no such file or directory, scandir 'C:\Users\xxx\AppData\Local\Temp\jest\ts-jest'
    at Object.fs.readdirSync (fs.js:913:18)
    at processResult (C:\Workspace\GitHub\TeBeCo\Webpack-seeds\react-typescript2-jest\node_modules\ts-jest\dist\coverageprocessor.js:20:26)
    at module.exports (C:\Workspace\GitHub\TeBeCo\Webpack-seeds\react-typescript2-jest\jestTestResultsProcessor.js:4:12)
    at source.getTestPaths.then.then.then.runResults (C:\Workspace\GitHub\TeBeCo\Webpack-seeds\react-typescript2-jest\node_modules\jest\node_modules\jest-cli\b
uild\runJest.js:110:58)
npm ERR! Test failed.  See above for more details.
```

the folder `C:\Users\xxx\AppData\Local\Temp\jest\` exist but not the subfolder `ts-jest`
if i create it manually the error disapear

I've change the `cacheDirectory` to point to a non existing folder inside  the repository
```
{
    "cacheDirectory": ".ts-jest-cache"
}
```

the folder is well create but again it DOES NOT create the sub folder `ts-jest` inside and print the same stacktrace

--- a/src/coverageprocessor.ts
+++ b/src/coverageprocessor.ts
@@ -7,7 +7,7 @@ const remap = require('remap-istanbul/lib/remap');
 const writeReport = require('remap-istanbul/lib/writeReport');
 const istanbulInstrument = require('istanbul-lib-instrument');
 import pickBy = require('lodash.pickby')
-import {getJestConfig} from './utils';
+import { getJestConfig } from './utils';
 const glob = require('glob-all');
 
 interface CoverageMap {
@@ -18,11 +18,11 @@ interface CoverageMap {
 }
 
 declare const global: {
-    __ts_coverage__cache__: {
-        coverageConfig: any;
-        sourceCache: any[];
-        coverageCollectFiles: any[];
-    }
+  __ts_coverage__cache__: {
+    coverageConfig: any;
+    sourceCache: any[];
+    coverageCollectFiles: any[];
+  }
 };
 
 // full type https://github.com/facebook/jest/blob/master/types/TestResult.js
@@ -37,6 +37,9 @@ function processResult(result: Result): Result {
   let coveredFiles = [];
 
   let basepath = path.join(jestConfig.cacheDirectory, '/ts-jest/');
+  if (!fs.existsSync(basepath)) {
+    fs.mkdirSync(basepath)
+  }
   let cachedFiles = fs.readdirSync(basepath);
   cachedFiles.map((p) => {
     let filename = new Buffer(p.replace(basepath, ''), 'base64').toString('utf8');
@@ -53,18 +56,18 @@ function processResult(result: Result): Result {
   };
 
   const coverageCollectFiles =
-      coverageConfig.collectCoverage &&
+    coverageConfig.collectCoverage &&
       jestConfig.testResultsProcessor &&
       jestConfig.collectCoverageFrom &&
       jestConfig.collectCoverageFrom.length ?
-            glob.sync(jestConfig.collectCoverageFrom).map(x => path.resolve(root, x)) : [];
+      glob.sync(jestConfig.collectCoverageFrom).map(x => path.resolve(root, x)) : [];
 
   if (!coverageConfig.collectCoverage) return result;
 
   let coverage;
   try {
     coverage = [pickBy(result.coverageMap.data, (_, fileName) => includes(coveredFiles, fileName))];
-  } catch(e) {
+  } catch (e) {
     return result;
   }
 
@@ -73,7 +76,7 @@ function processResult(result: Result): Result {
 
   // //generate 'empty' coverage against uncovered files.
   // //If source is non-ts passed by allowJS, return empty since not able to lookup from cache
-  const emptyCoverage = uncoveredFiles.map((x:string) => {
+  const emptyCoverage = uncoveredFiles.map((x: string) => {
     let ret = {};
     if (sourceCache[x]) {
       let instrumenter = istanbulInstrument.createInstrumenter();

--- a/tests/__helpers__/runJest.ts
+++ b/tests/__helpers__/runJest.ts
@@ -1,6 +1,6 @@
 // from: https://github.com/facebook/jest/blob/master/integration_tests/runJest.js
 
-const {fileExists} = require('./utils');
+const { fileExists } = require('./utils');
 const path = require('path');
 const spawnSync = require('cross-spawn').sync;
 

